File: baekjoon/11723/solution.js
function solution(n, rows) {
  const set = new Array(21).fill(0);
  let result = "";
  for (const [comm, x] of rows) {
    switch (comm) {
      case "add":
        set[x] = 1;
        break;
      case "remove":
        set[x] = 0;
        break;
      case "check":
        result += `${set[x]}\n`;
        break;
      case "toggle":
        set[x] = set[x] === 1 ? 0 : 1;
        break;
      case "all":
        for (let i = 1; i <= 20; i++) {
          set[i] = 1;
        }
        break;
      case "empty":
        for (let i = 1; i <= 20; i++) {
          set[i] = 0;
        }
        break;
    }
  }
  console.log(result);
}

//////
////
// input

const fs = require("fs");
const isLocal = process.platform !== "linux";
const filePath = isLocal ? "./input.txt" : "/dev/stdin";
const input = fs.readFileSync(filePath).toString().trim();

const cases = input.split("\n").filter((item) => !!item);

let idx = 0;
while (idx < cases.length) {
  const n = +cases[idx];
  const offset = 1;
  const rows = cases.slice(idx + offset, idx + n + offset).map((item) => {
    return item
      .trim()
      .split(" ")
      .map((it, i) => (i === 0 ? it : +it));
  });
  solution(n, rows);

  idx += n + offset;
}

File: baekjoon/11723/solution.py
import sys
# sys.stdin = open("input.txt", "r") 

n = int(sys.stdin.readline())

checked = [0] * 21
lines = []
for _ in range(n):
  input = sys.stdin.readline().rstrip().split(' ')

  if len(input) == 2:
    x = int(input[1])
    if input[0] == 'add':
      checked[x] = 1
    elif input[0] == 'remove':
      checked[x] = 0
    elif input[0] == 'check':
      print(checked[x])
    elif input[0] == 'toggle':
      checked[x] = 0 if checked[x] == 1 else 1
  else:
    if input[0] == 'all':
      checked = [1] * 21
    elif input[0] == 'empty':
      checked = [0] * 21